bundles/core/src/main/java/com/adobe/cq/commerce/demandware/preview/PreviewServiceImpl.java

```diff
@@ -159,42 +133,45 @@ private String getPreviewContent(final String endPoint, final Resource resource,
                 return renderedPreview;
             }
         }
-
+        
         // render AEM component for Demandware
         final String renderedComponentContent = renderService.render(resource, null, selectores);
-
-        // call Demandware to render preview for component
-        final HttpClientBuilder httpClientBuilder = clientProvider.getDefaultClient().getHttpClientBuilder();
+        final Page containingPage = getPage(resource);
+        DemandwareClient demandwareClient = getDemandwareClient(containingPage);
+        
+        final HttpClientBuilder httpClientBuilder = demandwareClient.getHttpClientBuilder();
+        
+        if (PropertiesUtil.toBoolean(previewServiceConfig.getStorfrontProtectionEnabled(), true)) {
+            credentialsProvider = createCredentialsProvider(previewServiceConfig, resource);
```

Comment:
you don't need to pass resource in order to get DemandwareConfig inside createCredentialsProvider method.
You already have DemandwareConfig, so just pass it instead of resource